api/src/controllers/publications.js

```diff
@@ -168,27 +170,46 @@ async function getGoogleScholarPublications(req, res) {
   const resultLinks = await page.$$('.gsc_a_t a');
   let links = [];
 
-  for (let i = noOfDummyLinks; i < resultLinks.length; i++) {
-    links.push(await resultLinks[i].getAttribute('href'));
-  }
+  if (resultLinks.length === noOfDummyLinks) {
+    // no pubs found
+    endOfProfile = true;
+    await browser.close();
 
-  await browser.close();
+    response = {
+      retrieved: publications.length,
+      newPublications: [],
+      reachedEnd: endOfProfile,
+    };
+  } else {
+    for (let i = noOfDummyLinks; i < resultLinks.length; i++) {
+      links.push(await resultLinks[i].getAttribute('href'));
+    }
 
-  await Promise.all(links.map((x) => scrapeGoogleScholar(x))).then((pub) =>
-    publications.push(...pub)
-  );
+    await browser.close();
 
-  console.timeEnd('timeToScrape');
+    await Promise.all(links.map((x) => scrapeGoogleScholar(x))).then((pub) =>
+      publications.push(...pub)
```

Comment:
end statements with `;`?

Reply:
If you're talking about line 191, then it doesn't need a `;` there cause it causes a syntax error, plus prettier doesn't put one there either.

Reply:
Sorry, I meant that `push` invocation on the list

Reply:
Woops, I misread the code. You're right. Resolving this comment.